docs/doxygen/collectives.md

```diff
@@ -1,20 +1,150 @@
+data elements. The
+"dt_elem_size" represents the size of the data element in bytes. The
+"num_participants" define the number of participants in the team. The data on
+the root is placed in the receive buffer ordered by the “ep” ordering. For
+example, if the participants’ endpoints are ordered as “ep\_a” to “ep\_n”, the
+data from the participant with ep_i is placed as an “ith” block on the receive
+buffer.
+
++ In the UCC\_COLL\_TYPE\_SCATTER operation, the root participant of the
+collective operation sends data to all other participants. It sends the same
+amount of data (block_size) to all participants. The size of the block
+(block_size) is “dt_elem_size * count”. The total amount of data received by the
+root is equal to block_size * num\_participants. Here, the “count” represents the
+number of data elements. The "dt_elem_size" represents the size of the data
+element in bytes. The "num_participants" define the number of participants in
+the team.
+
++ In the UCC\_COLL\_TYPE\_ALLTOALL collective operation, all participants
+exchange a fixed amount of the data. For a given participant, the size of data
+in src buffer is “size”, where size is dt\_elem\_size * count * num_participants.
+Here, the “count” represents the number of data elements. The "dt_elem_size"
+represents the size of the data element in bytes. The "num_participants" define
+the number of participants in the team. The size of src buffer is the same as
+the dest buffer, and it is the same across all participants. Each participant
+exchanges “dt\_elem\_size * count “ data with every participant of the collective.
+
++ In UCC\_COLL\_TYPE\_REDUCE collective the element-wise reduction operation is
+performed on the src buffer of all participants in the collective operation. The
+result is stored on the dst buffer of the root. The size of src buffer and dst
+buffer is the same, which is equal to “dt_elem_size * count”. Here, the “count”
+represents the number of data elements. The "dt_elem_size" represents the size
+of the data element in bytes.
+
++ The UCC\_COLL\_TYPE\_ALLREDUCE first performs an element-wise reduction on the
+src buffers of all participants. Then the result is distributed to all
+participants. After the operation, the results are available on the dst buffer
+of all participants. The size of src buffer and dst buffer is the same for all
+participants. The size of src buffer and dst buffer is the same, which is equal
+to “dt_elem_size * count”. Here, the “count” represents the number of data
+elements. The "dt_elem_size" represents the size of the data element in bytes.
+
++ The UCC\_COLL\_TYPE\_REDUCE\_SCATTER first performs a element-wise
+reduction on the src buffer and then scatters the result to the dst buffer. If
+the size of src buffer is “count * dt_elem_size”, where dt_elem_size is the
+number of bytes for the data type element and count is the number of elements of
+that datatype. Assuming the result can be divided into “n” blocks, the ith block
```

Comment:
For ReduceScatter we must agree:
1. Either we only support the case when count % team_size == 0 -> MPI case.
2. If we are to support other case then we need to define how the remainder (count % team_size) is distributed. Then we must document it. E.g., the remainder can be distributed evenly among first (count % team_size) ranks - ie, each of those ranks will get +1 element

Reply:
Do we have use case for 2?

Reply:
Discussion with Val: Let's support case 1.